### PR TITLE
K8S-2323: Fix bucket creation

### PR DESCRIPTION
**Pull request by loulbani**
This fix will allow us to avoid this error when we don't use Ephemral bucket in our helm chart

Error: template: couchbase-operator/templates/couchbase-bucket.yaml:19:8: executing "couchbase-operator/templates/couchbase-bucket.yaml" at <eq $spec.kind "CouchbaseEphemeralBucket">: error calling eq: incompatible types for comparison

### CODE REVIEW

File: charts/couchbase-operator/templates/couchbase-bucket.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ items:
   spec:
 
 {{/* Remove index replica for Ephemeral Type */}}
-{{- if (eq $spec.kind "CouchbaseEphemeralBucket")  -}}
+{{- if eq (default $spec.kind "CouchbaseBucket") "CouchbaseEphemeralBucket") -}}
```
Comment (tahmmee) [inline charts/couchbase-operator/templates/couchbase-bucket.yaml:21]: fix parenthesis here

**Review by tahmmee — COMMENTED** [review]
Thanks @loulbani, also when you are not using Ephemeral bucket, you can use buckets[].kind to set the bucket type which defaults to CouchbaseBucket.

File: charts/couchbase-operator/templates/couchbase-bucket.yaml (hunk 1 above)
Comment (loulbani) [reply charts/couchbase-operator/templates/couchbase-bucket.yaml]: Done

**Comment by tahmmee** [comment]
Looks good.  Last thing is update the Chart version to `2.2.006` so that the changes are published in a different release- thanks!

**Comment by patrick-stephens** [comment]
Apologies, had to push out a 2.2.006 update for an Openshift issue. If you update again to 2.2.007 I'll merge for you.

**Comment by loulbani** [comment]
> Apologies, had to push out a 2.2.006 update for an Openshift issue. If you update again to 2.2.007 I'll merge for you.

No worries, I just update the version as requested
Thanks!

**Comment by patrick-stephens** [comment]
Thanks again, all looks good to me (and @tahmmee obviously) so just going to get QE to quickly confirm and will release.

**Comment by patrick-stephens** [comment]
Just to clarify this is basically to ensure we set a default bucket type for the helm chart using the `kind` field which is not part of the Couchbase CRD. The issue could be triggered with the defaults like so:
```
helm install test couchbase/couchbase-operator --set buckets.default.kind=null
Error: template: couchbase-operator/templates/couchbase-bucket.yaml:21:8: executing "couchbase-operator/templates/couchbase-bucket.yaml" at <eq $spec.kind "CouchbaseEphemeralBucket">: error calling eq: incompatible types for comparison
```
Similarly if people add their own buckets then no default was provided and it would trigger the same issue.

**Comment by patrick-stephens** [comment]
Thanks again @loulbani for sorting this, there should now be a 2.2.007 release for this.
Let me know of any issues and ideally if it solves your problems so I can close K8S-2323 - I'll assume it is fine unless I hear otherwise but be good to get positive confirmation.

**Pull request merged**